### PR TITLE
Add relaxation kwarg to solve

### PR DESCRIPTION
**Pull request by joehuchette**
Ref #587, still open to a bikeshed to the naming convention if @IainNZ wants to chime in.

### CODE REVIEW

File: test/model.jl [hunk 1]
```diff
@@ -738,3 +738,23 @@ facts("[model] Test getValue on OneIndexedArrays") do
...
+    m = Model()
+    @defVar(m, 1 <= x <= 2, SemiCont)
+    @defVar(m, 1.5 <= y <= 2, Int)
+    @defVar(m, z, Bin)
+    @defVar(m, 0.5 <= w <= 0.5, Int)
+    @defVar(m, 1 <= v <= 2)
+    @defVar(m, -2 <= t <= -1, SemiInt)
+
+    @setObjective(m, Min, x + y + z + w + v - t)
+    solve(m, relaxation=true)
```
Comment (mlubin) [inline test/model.jl]: What happens if you call ``solve()`` again with ``relaxation=false``? Do we clear the relaxation data?

Comment (joehuchette) [inline test/model.jl]: Seems to work (returns infeasible with CPLEX).

Comment (mlubin) [inline test/model.jl]: For simplicity, what about setting ``internalModelLoaded=false`` after a solve with ``relaxation=true``?

Comment (joehuchette) [inline test/model.jl]: Works with me

File: test/model.jl [hunk 2]
```diff
@@ -738,3 +738,25 @@ facts("[model] Test getValue on OneIndexedArrays") do
...
+    @fact getValue(x) --> 0
+    @fact getValue(y) --> 1.5
+    @fact getValue(z) --> 0
+    @fact getValue(w) --> 0.5
+    @fact getValue(v) --> 1
+    @fact getValue(t) --> 0
+    @fact getObjectiveValue(m) --> 0 + 1.5 + 0 + 0.5 + 1 + 0
```
Comment (mlubin) [inline test/model.jl]: Could you add tests with some switching between ``true`` and ``false``?

**Comment by mlubin** [comment]
The name ``relaxation`` still doesn't sit well with me. It will definitely send some people to the docs asking "which relaxation?"

**Comment by joehuchette** [comment]
Not sure that's a bad thing, or avoidable, really. I'm warming up to ``continuous_relaxation``, though it doesn't fix the "lookup in the docs" issue.

**Comment by IainNZ** [comment]
I think there is no fixing the "lookup in docs" situation. But, of course, someone would have to look in the docs to find it in the first place. I'd say relaxing integrality is by the most common usage for people who use JuMP, and I don't think it'll be scrutinized heavily. If we get into more general relaxations later, we can cross that bridge then.

One option is `relaxation=:Continuous` I guess if we want to avoid booleans.

File: src/solvers.jl [hunk 3]
```diff
@@ -168,6 +170,8 @@ function solve(m::Model; suppress_warnings=false,
         end
     end
 
+    relaxation && (m.internalModelLoaded = false)
```
Comment (mlubin) [inline src/solvers.jl]: Maybe add a comment here like: ``# don't keep relaxed model in memory``? I had trouble reading this line on the first try.

**Comment by mlubin** [comment]
Let's just go with ``relaxation=true`` for now, can always deprecate later. Needs docs before merging.

File: doc/refmodel.rst [hunk 4]
```diff
@@ -43,7 +43,7 @@ Methods
...
-* ``solve(m::Model;  suppress_warnings=false)`` - solves the model using the selected solver (or a default for the problem class), and takes two optional arguments that are disabled by default. Setting ``suppress_warnings`` to ``true`` will suppress all JuMP-specific output (e.g. warnings about infeasibility and lack of dual information) but will not suppress solver output (which should be done by passing options to the solver).
+* ``solve(m::Model; suppress_warnings=false, relaxation=false)`` - solves the model using the selected solver (or a default for the problem class), and takes two optional arguments that are disabled by default. Setting ``suppress_warnings`` to ``true`` will suppress all JuMP-specific output (e.g. warnings about infeasibility and lack of dual information) but will not suppress solver output (which should be done by passing options to the solver). Setting ``relaxation=true`` solves the standard continuous relaxation for the model: that is, integrality is dropped, SOS constraints are not enforced, and semi-continuous and semi-integer variables with bounds ``[l,u]`` are replaced with bounds ``[min(l,0),max(u,0)]``.
```
Comment (mlubin) [inline doc/refmodel.rst]: Should clarify that SOS is special ordered set. Many people will read it as sum of squares.

**Pull request merged**